### PR TITLE
Add Encoder API to examples for gosdk, change function signatures in Encoder to match other Interface naming

### PR DESCRIPTION
**Pull request by randhid**
When adding the api helper example, I noticed the golang methods in Encoder weren't following the naming scheme of other components. Changed `GetProperties` to `Properties` and `GetPosition` to `Position`. My bad for not noticing earlier, but if this is overkill I can revert that change, and just use `GetProperties` in examples/api.

cc: @edaniels. mea culpa.

- [x] Test on bucket with single encoder
- [x] Test on archie with incremental encoder

### CODE REVIEW

File: components/encoder/client.go [hunk 1]
```diff
@@ -43,7 +43,7 @@ func NewClientFromConn(
 
 // GetPosition returns the current position in terms of ticks or
```
Comment (martha-johnston) [inline components/encoder/client.go:44]: Position

File: components/encoder/client.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func (c *client) ResetPosition(ctx context.Context, extra map[string]interface{}
 }
 
 // GetProperties returns a list of all the position types that are supported by a given encoder.
```
Comment (martha-johnston) [inline components/encoder/client.go:77]: change to `Properties`

Comment (martha-johnston) [inline components/encoder/client.go:77]: change to `Properties`

**Review by Kschappacher — APPROVED** [review]
The code samples encoder part looks good!

**Review by martha-johnston — COMMENTED** [review]
just those two comments and then LGTM!

**Review by stevebriskin — APPROVED** [review]

**Pull request merged**